crates/astria-sequencer/src/app.rs

```diff
@@ -293,13 +297,35 @@ impl App {
                 continue;
             };
 
-            // store transaction execution result, indexed by tx hash
             let tx_hash = Sha256::digest(&tx);
+
+            // Sum up the total bytes in sequence actions
+            let mut sequenced_data_size = 0;
+            for action in signed_tx.clone().into_unsigned().actions {
+                if let Some(sequence_action) = action.as_sequence() {
+                    sequenced_data_size += sequence_action.data.len();
+                }
+            }
+
+            // Don't include tx if it would make the sequenced block data too large.
+            if total_sequenced_data_size + sequenced_data_size > MAX_BLOCK_SEQUENCE_DATA_SIZE {
+                debug!(
+                    transaction_hash = %telemetry::display::hex(&tx_hash),
+                    included_data_bytes = total_sequenced_data_size,
+                    tx_data_bytes = sequenced_data_size,
+                    max_data_bytes = MAX_BLOCK_SEQUENCE_DATA_SIZE,
+                    "excluding transaction: max block sequenced data limit reached"
+                );
+                continue;
```

Comment:
I first wanted to suggest `break;`ing here, but in fact this just drops signed txes that don't fit the rest, but attempts to fill up the buffer as much as possible, correct?

Reply:
Correct, if there is a follow on smaller tx data that can fit we should grab it.

We are explicitly not going into the whole napsack problem, but a single too big tx shouldn't stop us fitting more.